### ISSUE TITLE
[8.0] hr_payroll can't change payslip date

### ISSUE THREAD

**Issue by tarzan0820** [odoo/odoo#8883]
After computing a payslip but without confirming, any attempt to change the date_from field results in the following error:

ValueError: "name 'date_from' is not defined" while evaluating
'[date_from, date_to, employee_id, contract_id]'

It looks like it is unable to evaluate the on_change function declared on the date_from field

**Comment by twinslady1612** [comment]
```     
#delete old worked days lines
        old_worked_days_ids = ids and worked_days_obj.search(cr, uid, [('payslip_id', '=', ids[0])], context=context) or False
        if old_worked_days_ids:
            worked_days_obj.unlink(cr, uid, old_worked_days_ids, context=context)

        #delete old input lines
        old_input_ids = ids and input_obj.search(cr, uid, [('payslip_id', '=', ids[0])], context=context) or False
        if old_input_ids:
            input_obj.unlink(cr, uid, old_input_ids, context=context)
```
It will be ok if you delete the source code above in hr_payroll/payslip.py/ function `onchange_employee_id`

**Comment by m3asmi** [comment]
I got the same issue when changing the empoyee,
can this be fixed in code ?

**Comment by leio** [comment]
This looks similar issue like https://github.com/odoo/odoo/issues/8883 solved for another module. unlink calls in an onchange method. Issue #3116 looks similar issue to this one here, perhaps duplicates.
Perhaps instead of deleting these lines as twinslady1612 above suggested, we also need to better send the (2, id) command tuples back like the voucher PR

**Comment by JKE-be** [comment]
Thanks for the reporting.
It has been merged in 8.0: commit odoo/odoo@d43d1ddf4b038ca0500770793b9153f82d1aa257
